### DISCUSSION TITLE
wakeupChannel_->setReadCallback error

### DISCUSSION THREAD

**Discussion by chanchann** [chenshuo/muduo#581] [General]
In https://github.com/chenshuo/muduo/blob/cpp11/muduo/net/EventLoop.cc

line 87 :   wakeupChannel_->setReadCallback(std::bind(&EventLoop::handleRead, this));

line 244 : void EventLoop::handleRead() ...

But In https://github.com/chenshuo/muduo/blob/cpp11/muduo/net/Channel.h

line 37 : typedef std::function<void(Timestamp)> ReadEventCallback; 

The ReadEventCallback has one parameter. If we call this callback, then a small error occurs.

**Comment by chanchann** [comment]
We can just add the placeholder : 

wakeupChannel_->setReadCallback(std::bind(&EventLoop::handleRead, this, std::placeholders::_1));

And add the parameter although we don't need it :

void EventLoop::handleRead(Timestamp receiveTime);

**Comment by chenshuo** [comment]
It's not an error, but a feature of std::bind().

Ref. docs:

1. https://www.boost.org/doc/libs/1_66_0/libs/bind/doc/html/bind.html#bind.purpose.using_bind_with_functions_and_fu
> **Any extra arguments are silently ignored**, just like the first and the second argument are ignored in the third example.

2. https://en.cppreference.com/w/cpp/utility/functional/bind
> If some of the arguments that are supplied in the call to g() are not matched by any placeholders stored in g, **the unused arguments are evaluated and discarded**.

**Comment by chenshuo** [comment]
Close as WAI (working as intended), dup of #456.
